src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java

```diff
@@ -187,15 +180,10 @@ public void testConnection() {
                 return action.apply(sftp);
             }
         } catch (IOException exc) {
-            throw new FtpException("Unable to upload file.", exc);
-        } finally {
-            try {
-                if (ssh != null) {
-                    ssh.disconnect();
-                }
-            } catch (IOException e) {
-                logger.warn("Error closing ssh connection.", e);
```

Comment:
looks like behaviour is now changed.
before: logging a warning
now: throws `FtpException`.

Was this intended?

Reply:
Closing ssh client is basically a failure. This code was, my guess, done because it was not closeable in the first place. Hence

```java
SSHClient client = null;

try {
    ...
} catch (IOException exc) {
    try closing
}
```

any IOException before closing is still caught and re-thrown with correct/better name inside

Reply:
Throwing an exception can result in rolling back a transaction for example.
I assume the exception from closing connection was not rethrown and instead just logged as warning before for a reason.

Reply:
Made compatible with https://github.com/hmcts/send-letter-service/pull/370/commits/8942b7851060a9aeed4ff3ea94021e1fa8865c90. It should've used try with resources from the beginning but since it is the topic, making necessary changes now. Didn't check but could have been not yet `Closeable` at the time of usage